src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -616,7 +615,7 @@ private BoundExpression CheckValue(BoundExpression expr, BindValueKind valueKind
                 var methodGroup = (BoundMethodGroup)expr;
                 CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
                 var resolution = this.ResolveMethodGroup(methodGroup, analyzedArguments: null, useSiteInfo: ref useSiteInfo, options: OverloadResolution.Options.None);
-                Debug.Assert(!resolution.IsNonMethodExtensionMember(out _));
+                Debug.Assert(!resolution.IsNonMethodExtensionMember(out _)); // expr wouldn't be a method group in the first place if it resolved to a non-method extension member
```

Comment:
>expr wouldn't be a method group in the first place if it resolved to a non-method extension member [](http://example.com/codeflow?start=80&length=98)

Sorry, I do not follow this argument. Don't we always start with a MethodGroup when we resolve extensions? The same question is for other places where the same comment is added.

Reply:
The only expression that can produce a non-method extension member access is a member access. When we bind a member access, we'll already have done `ResolveExtensionMemberAccessIfResultIsNonMethod` and resolved to non-method extension member. The method group only survives until `CheckValue` if it's not a non-member extension member.